### PR TITLE
refactor(docker-api): Update Markdown request handling for multiple URLs and performance metrics

### PR DESCRIPTION
**Pull request by prokopis3**
## Summary
This pull request updates the Markdown endpoint in the Docker API to support multiple URLs per request, improves performance tracking, and enhances input validation. It also updates the frontend playground to support multi-URL input and introduces comprehensive asynchronous tests to ensure reliability of the /md endpoint.

## List of files changed and why
* `server.py` – Integrated performance logging and ensured the server-side logic aligns with the new MarkdownRequest schema and response structure.

* `schemas.py` – Updated the MarkdownRequest model to accept and validate a list of URLs with a specified limit.

* `api.py` – Modified the handle_markdown_request function to process multiple URLs asynchronously using the browser pool system, returning results and performance data.

* `test_markdown.py` – Added async tests covering multiple scenarios: valid/invalid inputs, empty arrays, and error cases, multiple urls.

* `index.html` – Updated the playground interface to allow submitting multiple URLs from the textarea to the /md endpoint.

## How Has This Been Tested?

* The Docker server was run locally and the playground interface was manually tested to verify it correctly handles multiple URLs for the /md endpoint.

* All tests in test_markdown.py were executed and passed successfully.

* Additional unit tests were added for validation, error handling, and markdown extraction performance.

* Verified edge cases such as empty input, malformed URLs, and large payloads.

docker-mds-api_logs
docker-mds-api_playground
docker-mds-api_tests

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added/updated unit tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by ntohidi** [comment]
@coderabbitai review

File: tests/docker/test_markdown.py [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+import pytest
+import httpx
+import asyncio
+
+class FilterType(str, Enum):
+    RAW = "raw"
+    FIT = "fit"
+    BM25 = "bm25"
+    LLM = "llm"
+
```
[the remark this answers is not in the document]
Comment (prokopis3) [reply tests/docker/test_markdown.py:11]: ❗ Import Issue After Removing FilterType from Test File
```python
from deploy.docker.utils import FilterType
```

running pytest results in the following error:

```python
ModuleNotFoundError: No module named 'deploy'
```

I also tried a relative import like:

```python
from ...deploy.docker.utils import FilterType
```

But that fails too with

```python
ImportError: attempted relative import with no known parent package
```

[the remark this answers is not in the document]
Comment (prokopis3) [reply tests/docker/test_markdown.py:11]: I’d prefer not to modify the project structure right now by adding __init__.py files.
So let’s keep the initial approach and define the FilterType enum locally within the tests.
That works fine for now, even if it duplicates the definition a bit.

**Comment by prokopis3** [comment]
@ntohidi I’ve pushed the fixes to the PR — let me know if it looks okay now!

**Comment by ntohidi** [comment]
> @ntohidi I’ve pushed the fixes to the PR — let me know if it looks okay now!



> @ntohidi I’ve pushed the fixes to the PR — let me know if it looks okay now!

sure, i'll check!
